x/auth/vesting/types/vesting_account_test.go

```diff
@@ -868,134 +862,99 @@ func TestClawback(t *testing.T) {
-	expectedDestCoins := sdk.Coins{sdk.NewInt64Coin(feeDenom, 600), sdk.NewInt64Coin(stakeDenom, 50)}
-	require.Equal(t, expectedDestCoins, destBalances)
-	// none of the remaining addr coins should be spendable
-	require.True(t, app.BankKeeper.SpendableCoins(ctx, addr).Empty())
-
-	// at hour 12, the unlock should finish and the account should have its 400fee, 50 stake liquid
-	ctx = ctx.WithBlockTime(now.Add(15 * time.Hour))
-	va2 = app.AccountKeeper.GetAccount(ctx, addr).(*types.ClawbackVestingAccount)
-	require.Equal(t, expectedAddrCoins, app.BankKeeper.SpendableCoins(ctx, addr))
-}
-
-func TestCliffClawback(t *testing.T) {
-	c := sdk.NewCoins
-	fee := func(x int64) sdk.Coin { return sdk.NewInt64Coin(feeDenom, x) }
-	stake := func(x int64) sdk.Coin { return sdk.NewInt64Coin(stakeDenom, x) }
-	now := tmtime.Now()
-
-	// set up simapp and validators
-	app := simapp.Setup(false)
-	ctx := app.BaseApp.NewContext(false, tmproto.Header{}).WithBlockTime((now))
-	valAddr, val := createValidator(t, ctx, app, 100)
-	require.Equal(t, "stake", app.StakingKeeper.BondDenom(ctx))
-
-	totalVestingCoins := sdk.NewCoins(sdk.NewInt64Coin(feeDenom, 1000), sdk.NewInt64Coin(stakeDenom, 100))
-	lockupPeriods := []types.Period{
-		{Length: int64(12 * 3600), Amount: c(fee(1000), stake(100))},
-	}
-	vestingPeriods := []types.Period{
-		{Length: int64(8 * 3600), Amount: c(fee(200))},            // 8am
-		{Length: int64(1 * 3600), Amount: c(fee(200), stake(50))}, // 9am
-		{Length: int64(6 * 3600), Amount: c(fee(200), stake(50))}, // 3pm
-		{Length: int64(2 * 3600), Amount: c(fee(200))},            // 5pm
-		{Length: int64(1 * 3600), Amount: c(fee(200))},            // 6pm
+	// each test creates a new clawback vesting account, with the lockup and vesting periods defined above.
+	// the clawback is executed at the test case's provided time, and expects that post clawback,
+	// the address has a total of `vestingAccBalance` coins, but only `spendableCoins` are spendable.
+	// It expects the clawback acct funder to have `funderBalance` (aka that amt clawed back)
+	testCases := []struct {
+		name              string
+		ctxTime           time.Time
+		vestingAccBalance sdk.Coins
+		spendableCoins    sdk.Coins
+		funderBalance     sdk.Coins
+	}{
+		{
+			"clawback before all vesting periods, before cliff ended",
+			now.Add(7 * time.Hour),
+			// vesting account should not have funds after clawback
+			sdk.NewCoins(),
+			sdk.Coins(nil),
+			// all funds should be returned to funder account
+			sdk.NewCoins(sdk.NewCoin(feeDenom, sdk.NewInt(1000)), sdk.NewCoin(stakeDenom, sdk.NewInt(100))),
+		},
+		// fails
+		{
+			"clawback after two vesting periods, before cliff ended",
+			now.Add(10 * time.Hour),
+			// vesting account should not have funds after clawback
+			sdk.NewCoins(),
+			sdk.Coins(nil),
+			// all funds should be returned to funder account
+			sdk.NewCoins(sdk.NewCoin(feeDenom, sdk.NewInt(1000)), sdk.NewCoin(stakeDenom, sdk.NewInt(100))),
+		},
```

Comment:
Actually this test case is wrong! After two vesting periods, the vested tokens should be in the account, but should be locked!